dbms/programs/server/HTTPHandler.cpp

```diff
@@ -407,11 +407,11 @@ void HTTPHandler::processQuery(
     {
         if (http_request_compression_method_str == "gzip")
         {
-            in_post = std::make_unique<ZlibInflatingReadBuffer>(*in_post_raw, CompressionMethod::Gzip);
+            in_post = std::make_unique<ZlibInflatingReadBuffer>(std::move(in_post_raw), CompressionMethod::Gzip);
         }
         else if (http_request_compression_method_str == "deflate")
         {
-            in_post = std::make_unique<ZlibInflatingReadBuffer>(*in_post_raw, CompressionMethod::Zlib);
+            in_post = std::make_unique<ZlibInflatingReadBuffer>(std::move(in_post_raw), CompressionMethod::Zlib);
         }
 #if USE_BROTLI
```

Comment:
Why don't you use *std::move* for *BrotliReadBuffer* ? And Is it really necessary to move *in_post_raw* inside constructor?

Reply:
> Why don't you use _std::move_ for _BrotliReadBuffer_ ? 
Its mistake, fixed.
> And Is it really necessary to move _in_post_raw_ inside constructor?
Yes, we need it because in_post_raw it is local pointer.